templates/lib/tasks/scanning.rake

```diff
@@ -7,4 +7,58 @@ task :brakeman do
   end
 end
 
-task default: "brakeman"
+namespace :bundler do
+  begin
+    require 'bundler/audit/cli'
+
+    desc 'Updates the ruby-advisory-db and runs audit'
+    task :audit do
+      %w(update check).each do |command|
+        Bundler::Audit::CLI.start [command]
+      end
+    end
+  rescue LoadError
+    # no-op, probably in a production environment
+  end
+end
+
+namespace :yarn do
+  desc "Run yarn audit"
+  task :audit do
+    stdout, stderr, status = Open3.capture3("yarn audit --json")
+    unless status.success?
+      puts stderr
+      parsed = JSON.parse("[#{stdout.lines.join(",")}]")
+      puts JSON.pretty_generate(parsed)
+      if /503 Service Unavailable/.match?(stderr)
+        puts "Ignoring unavailable server"
+      elsif all_issues_ignored?(parsed)
+        puts "Ignoring known and accepted yarn audit results"
+      else
+        puts "Failed with exit code #{status.exitstatus}"
+        exit status.exitstatus
+      end
+    end
+  end
+end
+
+def all_issues_ignored?(issues)
+  present_advisories_with_frequencies = Hash.new { |hash, key| hash[key] = 0 }
+
+  # Only look at audit advisories, and not audit summaries
+  issues.select{|issue_json| issue_json['type'] == 'auditAdvisory'}.each do |issue_json|
+    present_advisories_with_frequencies[issue_json['data']['advisory']['id']] += 1
+  end
+
+  # Advisory ID to be ignored with number of times it appears in project dependencies
+  # And, a comment as to why we're ignoring
+  ignored_advisories_with_frequencies = {
+    # 1005154 => 2, # high - inefficient regex in dev server and at build time
```

Comment:
[discussion] When generating a new project, this currently fails for the same reason it did in the Pangolin repo (three advisories). I think we should encourage each project to review any failures and manually add their own ignores to start, but this means that this rake task would fail currently out of the box. What do you think about this approach?

Reply:
I definitely agree that each project should be inspecting the failures and deciding for themselves if they're acceptable or need to be mitigated in some way.

Even if we wanted to accept risk on behalf of new apps, I don't think we'd be able to reliably keep up with changing findings.